libs-scala/resources/src/main/scala/com/digitalasset/resources/HasExecutionContext.scala

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.resources
+
+import scala.concurrent.ExecutionContext
+
+trait HasExecutionContext[Context] {
+  def executionContext(context: Context): ExecutionContext
+}
+
+object HasExecutionContext {
+  def apply[Context: HasExecutionContext]: HasExecutionContext[Context] =
+    Option(implicitly[HasExecutionContext[Context]]).get
```

Comment:
`implicitly` looks neat, but I would make the evidence explicit and just use that.